src/test/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilterUnitTest.java

```diff
@@ -0,0 +1,427 @@
+
+    private final ClassifierAccuracyData classifierAccuracyData = new ClassifierAccuracyData(testAccuracyDataJsonFile);
+    private final double[] predictYProbaSerial = predictProba(
+            XGBoostEvidenceFilter.loadPredictor(localClassifierModelFile), classifierAccuracyData.features
+    );
+    private static final FeaturesTestData featuresTestData = new FeaturesTestData(testFeaturesJsonFile);
+
+    private static final StructuralVariationDiscoveryArgumentCollection.FindBreakpointEvidenceSparkArgumentCollection params
+            = new StructuralVariationDiscoveryArgumentCollection.FindBreakpointEvidenceSparkArgumentCollection();
+
+    private static final SAMFileHeader artificialSamHeader = initSAMFileHeader();
+    private static final String readGroupName = "Pond-Testing";
+    private static final String DEFAULT_SAMPLE_NAME = "SampleX";
+    public static final ReadMetadata readMetadata = initMetadata();
+    private static final PartitionCrossingChecker emptyCrossingChecker = new PartitionCrossingChecker();
+    private final List<BreakpointEvidence> evidenceList = Arrays.stream(featuresTestData.stringReps)
+            .map(strRep -> BreakpointEvidence.fromStringRep(strRep, readMetadata)).collect(Collectors.toList());
+
+    private static SAMFileHeader initSAMFileHeader() {
+        final SAMFileHeader samHeader = createArtificialSamHeader();
+        SAMReadGroupRecord readGroup = new SAMReadGroupRecord(readGroupName);
+        readGroup.setSample(DEFAULT_SAMPLE_NAME);
+        samHeader.addReadGroup(readGroup);
+        return samHeader;
+    }
+
+    /**
+     * Create synthetic SAM Header comptible with genome tracts (e.g. has all the primary contigs)
+     * Use genome gaps file to enforce this.
+     */
+    public static SAMFileHeader createArtificialSamHeader() {
+        Comparator<String> contigComparator = new Comparator<String>() {
+            @Override public int compare(String s1, String s2) {
+                return getKey(s1).compareTo(getKey(s2));
+            }
+
+            String getKey(final String contig) {
+                try {
+                    final int contigNum = Integer.parseInt(contig.substring(3));
+                    return "chr" + (contigNum < 10 ? "0" + contigNum : contigNum);
+                } catch(NumberFormatException e) {
+                    if(contig.equals("chrX") || contig.equals("chrY")) {
+                        return contig;
+                    } else {
+                        return "zzz" + contig;
+                    }
+                }
+            }
+        };
+
+        SortedMap<String,Integer> contigLengthMap = new TreeMap<>(contigComparator);
+        final String svGenomeGapsFile = params.svGenomeGapsFile;
+        int lineNumber = -1;
+        try(final InputStream inputStream = XGBoostEvidenceFilter.getInputStream(svGenomeGapsFile)) {
+            final BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
```

Comment:
It should only very rarely be necessary to write a custom parser in the GATK.  The lowest level one usually needs is to implement a `TableReader`.